kivakit-configuration/src/main/java/com/telenav/kivakit/configuration/Component.java

```diff
@@ -0,0 +1,237 @@
+ * {@link Settings} for details.
+ * </p>
+ *
+ * <ul>
+ *     <li>{@link #require(Class)} - Gets the given settings object or fails</li>
+ *     <li>{@link #require(Class, Enum)} - Gets the settings object of the given instance or fails</li>
+ *     <li>{@link #require(Class, String)} - Gets the settings object of the given instance or fails</li>
+ *     <li>{@link #require(Class, InstanceIdentifier)} - Gets the settings object of the given instance or fails</li>
+ *     <li>{@link #settings(Class)} - A settings object of the specified class or null if none exists</li>
+ *     <li>{@link #settings(Class, Enum)} - Get any given instance of the given settings object type</li>
+ *     <li>{@link #settings(Class, String)} - Get any given instance of the given settings object type</li>
+ *     <li>{@link #settings(Class, InstanceIdentifier)} - Get any given instance of the given settings object type</li>
+ *     <li>{@link #settingsRegistry()} - Retrieves the {@link Settings} registry for this component</li>
+ *     <li>{@link #registerDeployment(Deployment)} - Adds the settings objects for the given deployment</li>
+ *     <li>{@link #registerSettings(Settings)} - Adds the settings objects in the given settings registry</li>
+ *     <li>{@link #registerSettingsObject(Object)}  - Adds the given settings object</li>
+ *     <li>{@link #registerSettingsObject(Object, InstanceIdentifier)} - Adds the given settings object under the given instance identifier</li>
+ *     <li>{@link #registerSettingsIn(Folder)} - Adds settings objects from the given folder</li>
+ *     <li>{@link #registerSettingsIn(Package)} - Adds settings objects from the given package</li>
+ * </ul>
+ *
+ * <p><b>Object Lookup Registry</b></p>
+ *
+ * <p>
+ * Access to this component's lookup {@link Registry} is provided and fulfills basic needs for object wiring:
+ * </p>
+ *
+ * <ul>
+ *     <li>{@link #lookup(Class)} - Lookup the registered object of the given type, if any</li>
+ *     <li>{@link #lookup(Class, String)} - Find the given instance of the given class, if any</li>
+ *     <li>{@link #lookup(Class, Enum)} - Find the given instance of the given class, if any</li>
+ *     <li>{@link #lookup(Class, InstanceIdentifier)} - Find the given instance of the given class, if any</li>
+ *     <li>{@link #registerObject(Object)} - Register the given singleton object for lookup</li>
+ *     <li>{@link #registerObject(Object, String)} - Register the given instance of the given object for lookup</li>
+ *     <li>{@link #registerObject(Object, Enum)} - Register the given instance of the given object for lookup</li>
+ *     <li>{@link #registerObject(Object, InstanceIdentifier)} - Register the given instance of the given object for lookup</li>
+ *     <li>{@link #registry()} - Retrieves the lookup {@link Registry} for this component</li>
+ * </ul>
+ *
+ * @author jonathanl (shibo)
+ * @see Registry
+ * @see Settings
+ * @see Repeater
+ */
+public interface Component extends Repeater
+{
```

Comment:
BaseComponent also extends BaseRepeater, including its interface Repeater. I am thinking is Component necessary to extends Repeater

Reply:
I think it's clearer this way because ComponentMixin also needs the Repeater interface.

Reply:
I think also you could get a Component interface (which could be either a BaseComponent or a ComponentMixin) and you might want to use it as a Listener. This would require extending the Repeater interface.